src/shared/utils/cleanText.ts

```diff
@@ -6,15 +6,51 @@ import deepFreeze from 'deep-freeze';
 import sanitizeHtml, { defaults, IOptions, Attributes } from 'sanitize-html';
 import styles from './styles/clean-text.module.scss';
 
+type HeadingLevels = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'strong';
+const headingRegexp = /h(\d)/i;
+
+const getLowerHeadingLevel = (
+  initialLevel: HeadingLevels,
+  currentLevel: HeadingLevels
+) => {
+  const initialMatch = initialLevel.match(headingRegexp);
+  const currentMatch = currentLevel.match(headingRegexp);
+  if (initialMatch && currentMatch) {
+    const initialLevel = parseInt(initialMatch[1], 10);
+    const currentLevel = parseInt(currentMatch[1], 10);
+    return initialLevel + currentLevel <= 6
+      ? `h${initialLevel + currentLevel}`
+      : 'strong';
+  }
+  return 'strong';
+};
+
 // List of tags to remove from the library default accepted set
 const excludedTags = new Set(['a']);
 
-const headingToStrong = (_: string, attribs: Attributes) => ({
-  tagName: 'strong',
-  attribs: {
-    id: attribs.id,
-    class: styles.heading,
-  },
+const updateHeadingLevel =
+  (currentHLevel: HeadingLevels = 'h1') =>
```

Comment:
```suggestion
  (currentHLevel: HeadingLevels = 'strong') =>
```

Reply:
There's only one place where we control (more or less) the content, the heading levels and their consistency. All the other places we need to assume it's unsafe, so let's handle unsafe heading levels by default, and explicitely set the relative heading level where we know we're OK to do so. It means simpler/leaner code for unsafe bits, and in entry we're already opting-in (l.68) to a specific heading level by saying we already have a h1 just before the content.

Reply:
See below for the other corresponding changes to remove code complexity

Reply:
It really depends what we mean by `cleanText`. To me, it's not obvious that it's going to change all headings to `strong` (and likewise with the current implementation that `h1` is the current default). I would err towards an explicit definition with default being "don't touch headings". More verbose in use, but easier to understand for a dev who sees that code for the first time.

Reply:
Yes, definitely.